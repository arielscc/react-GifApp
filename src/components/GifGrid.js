import React from 'react';
import GridGifItem from './GridGifItem';
import { useFetchGifs } from '../hooks/useFetchGifs';
import { BallTriangle } from 'svg-loaders-react';
import PropTypes from 'prop-types';

function GifGrid({ category }) {
  const { data: images, loading } = useFetchGifs(category);

  // useEffect( () => {
  //   getGifs( category ).then( setImages )
  // }, [category])

  return (
    <div className="container__grid-gif">
      <h3 className="container-title animate__animated animate__flipInX">
        {' '}
        {category.toUpperCase()}{' '}
      </h3>

      <div className="loader">
        {loading && <BallTriangle stroke="#2B6CB0" />}
      </div>

      <div className="container">
        {images.map((item) => (
          <GridGifItem key={item.id} {...item} />
        ))}
      </div>
    </div>
  );
}

GifGrid.propTypes = {
  category: PropTypes.string.isRequired,
};

export default GifGrid;
