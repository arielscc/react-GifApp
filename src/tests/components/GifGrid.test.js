import GifGrid  from "../../components/GifGrid";
import React from 'react'

import { shallow } from "enzyme";
import '@testing-library/jest-dom'
import { useFetchGifs } from "../../hooks/useFetchGifs";

jest.mock('../../hooks/useFetchGifs');




describe('Pruebas del componente <GifGrid/>', () => {
  const category = 'Dragon Ball';
  // let wrapper = shallow( <GifGrid  category = { category }/>);
  // beforeEach( () => {
  //   wrapper = shallow( <GifGrid  category = { category }/>);
  // })
  
  test('Comparando con el snapshot', () => {
    
    useFetchGifs.mockReturnValue({
      data: [],
      loading: true
    })
    const wrapper = shallow( <GifGrid  category = { category }/>);
    expect( wrapper ).toMatchSnapshot()
  });
  
  test('Debe de mostrar items cuando se cargan imagenes de useFetchGrid', () => {
    const gifs = [{
      id: 'abc',
      url: 'http://alguna-imagen.net/imagen.gif',
      title: 'cualquier cosa'
    }]
    useFetchGifs.mockReturnValue({
      data: gifs,
      loading: false
    })

    const wrapper = shallow( <GifGrid  category = { category }/>);
    expect( wrapper ).toMatchSnapshot()
    
  });
});