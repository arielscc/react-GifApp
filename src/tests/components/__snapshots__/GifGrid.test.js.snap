// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas del componente <GifGrid/> Comparando con el snapshot 1`] = `
<div
  className="container__grid-gif"
>
  <h3
    className="container-title animate__animated animate__flipInX"
  >
     
    DRAGON BALL
     
  </h3>
  <div
    className="loader"
  >
    <m
      stroke="#2B6CB0"
    />
  </div>
  <div
    className="container"
  />
</div>
`;

exports[`Pruebas del componente <GifGrid/> Debe de mostrar items cuando se cargan imagenes de useFetchGrid 1`] = `
<div
  className="container__grid-gif"
>
  <h3
    className="container-title animate__animated animate__flipInX"
  >
     
    DRAGON BALL
     
  </h3>
  <div
    className="loader"
  />
  <div
    className="container"
  >
    <GridGifItem
      id="abc"
      key="abc"
      title="cualquier cosa"
      url="http://alguna-imagen.net/imagen.gif"
    />
    <GridGifItem
      id="a12312"
      key="a12312"
      title="cualquier cosa"
      url="http://alguna-imagen.net/imagen.gif"
    />
  </div>
</div>
`;
